### PR TITLE
Add Folder meta data logic.

### PR DESCRIPTION
**Pull request by doncung**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @doncung has signed the CLA. Thanks for the pull request!

**Comment by doncung** [comment]
Still need to do add unit tests.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java [hunk 1]
```diff
@@ -103,43 +144,145 @@ protected String getFilesUrl() {
...
+
+    /**
+     * Gets a request that deletes the metadata for a specific template on a folder
+     *
+     * @param id    id of the folder to retrieve metadata for
+     * @param template    metadata template to use
+     * @return  request to delete metadata on a folder
+     */
+    public BoxRequestsMetadata.DeleteItemMetadata getDeleteFolderMetadataTemplateRequest(String id, String template) {
+        BoxRequestsMetadata.DeleteItemMetadata request = new BoxRequestsMetadata.DeleteItemMetadata(getFileMetadataUrl(id, template), mSession);
```
Comment (rgcorrea) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java]: Please correct to "getFolderMetadataUrl"

File: box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java [hunk 2]
```diff
@@ -103,43 +144,145 @@ protected String getFilesUrl() {
...
+    /**
      * Gets a request that deletes the metadata for a specific template on a file
      *
      * @param id    id of the file to retrieve metadata for
      * @param template    metadata template to use
      * @return  request to delete metadata on a file
      */
+    @Deprecated
     public BoxRequestsMetadata.DeleteFileMetadata getDeleteMetadataTemplateRequest(String id, String template) {
         BoxRequestsMetadata.DeleteFileMetadata request = new BoxRequestsMetadata.DeleteFileMetadata(getFileMetadataUrl(id, template), mSession);
```
Comment (rgcorrea) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java]: Can you re-route this to the new API ? Then we avoid duplicating this code.

File: box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java [hunk 3]
```diff
@@ -103,43 +144,145 @@ protected String getFilesUrl() {
...
+    /**
      * Gets a request that updates the metadata for a specific template on a file
      *
      * @param id    id of the file to retrieve metadata for
      * @param scope    currently only global and enterprise scopes are supported
      * @param template    metadata template to use
      * @return  request to update metadata on a file
      */
+    @Deprecated
     public BoxRequestsMetadata.UpdateFileMetadata getUpdateMetadataRequest(String id, String scope, String template) {
```
Comment (rgcorrea) [inline box-content-sdk/src/main/java/com/box/androidsdk/content/BoxApiMetadata.java]: I believe re-routing this to the new API is cleaner.

**Pull request merged**